[MRG+1] HOTFIX for incorrect cast in liblinear.pyx

MechCoder opened the pull request.

MechCoder committed:
HOTFIX for incorrect cast in liblinear.pyx

MechCoder commented:
@amueller sorry, I did not check after @larsmans told me that the memoryview type should be float

agramfort commented:
what was failing?

amueller commented:
@agramfort the C code was not matching the cython code. The cython code did not compile.

amueller commented:
I'm not familiar with the code so it is really hard for me to review.

MechCoder commented:
https://github.com/scikit-learn/scikit-learn/pull/5274#discussion-diff-42377195

amueller commented:
rebase?

amueller commented:
I messed with master so you may need to reset. I'll buy you a beer when I'm back ;)

ogrisel commented:
I don't understand this hotfix, why revert to `char*`?

ogrisel commented:
Can you please remove the merge commit? create a new branch off of the current `upstream/master` and just cherry-pick  fdac94b into it and force push to `MechCoder:hot_fix`.

MechCoder commented:
As far as the hotfix goes, I do not have time to investigate further right now (exams approaching) but while rerunning `cython liblinear.pyx` I got this error https://github.com/scikit-learn/scikit-learn/pull/5274#issuecomment-150589899 , which somehow looks like numpy returns a char pointer for a data buffer in NumPy, which cython is not able to implicitly cast into a double pointer. (I'm just speculating not sure though)

@amueller Thanks for the offer ! But I am not much of a beer fan and would prefer either rum or whisky :)

ogrisel commented:
The content of b3ab377 is not the cython cast hotfix for the liblinear sample weights.

MechCoder commented:
just a second, fixing it up

ogrisel commented:
fdac94b0da682e617161a3c05cb47ee7133dd931 has the right fix in case you lost it.

ogrisel commented:
Ok for using the `char*` type, it seems to be a Cython / numpy thingy. At least it's consistent with the rest of the file.

ogrisel commented:
I cancelled the travis build because 25d50a6 still does not have the correct fix. You should use `git log -p` to check the content of the commit before pushing if you are not sure.

MechCoder commented:
Sorry about the trouble, it should look all right now.

I tried the cherry-pick approach and it did not seem to work. Hence I reverted each of the faulty commits that seemed to show up and squashed.

amueller commented:
ping @larsmans @agramfort ?

ogrisel commented:
+1 if travis is green.

ogrisel commented:
travis is green, merging.

arthurmensch commented:
I ll rebase my PR ASAP
Le 23 oct. 2015 20:10, "Olivier Grisel" <notifications@github.com> a écrit :

> travis is green, merging.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/5565#issuecomment-150651272>
> .
>

MechCoder commented:
Sorry for the mess everyone should have checked twice :((

amueller commented:
No worries @MechCoder, happens to the best of us ;) @arthurmensch's PR will make sure we won't have this issue in the future.
